src/crimson/osd/state.h

```diff
@@ -10,7 +10,12 @@
 
 class OSDMap;
 
-class OSDState {
+namespace crimson::osd {
+
+// seastar::sharded puts start_single on core 0
+constexpr core_id_t PRIMARY_CORE = 0;
+
```

Comment:
It appears that you have introduced some slightly odd rules for which methods are valid on which core.  I think this implementation maintains all states on PRIMARY_CORE, but only ACTIVE and STOPPING on other cores.  Let's add a comment explaining that restriction:

```
/**
 * OSDState
 * 
 * Maintains state representing the OSD's progress from booting through
 * shutdown.
 *
 * Shards other than PRIMARY_CORE may use their local instance to check
 * on ACTIVE and STOPPING.  All other methods are restricted to
 * PRIMARY_CORE (such methods start with an assert to this effect).
 */
```